dev_docs/key_concepts/security.mdx

```diff
@@ -0,0 +1,101 @@
+---
+
+Security is everyone's responsibility. This is inclusive of design, product, and engineering. The purpose of this guide is to give a high-level overview of security constructs and expectations.
+
+This guide covers the following topics:
+
+* [API authorization](#api-authorization)
+* [The `kibana_system` user](#the-kibana_system-user)
+
+## API authorization
+Kibana API routes do not have any authorization checks applied by default. This means that your APIs are accessible to anyone with valid credentials, regardless of their permissions. This includes users with no roles assigned.
+This on its own is insufficient, and care must be taken to ensure that only authorized users can invoke your endpoints.
+
+### Adding API authorization
+Kibana leverages <DocLink id="kibDevDocsSavedObjectsIntro" text="Saved Objects" /> for a majority of its persistence. The Saved Objects Service performs its own authorization checks, so if your API route is primarily a CRUD interface to Saved Objects, then your authorization needs are already met.
+This is also true for derivatives of the Saved Objects Service, such as the Alerting and Cases services.
+
+If your endpoint is not a CRUD interface to Saved Objects, then your route should include `access` tags to ensure that only authorized users can invoke your endpoint. This is **especially** important if your route does any of the following:
+1. Performs non-insignificant processing, causing load on the Kibana server.
+2. Calls Elasticsearch using the internal `kibana_system` user.
+3. Calls a third-party service.
+4. Returns any non-public information to the caller, such as system configuration or state.
+
+More information on adding `access` tags to your routes can be found temporarily in the [legacy documentation](https://www.elastic.co/guide/en/kibana/current/development-security.html#development-plugin-feature-registration)
+
+### Why not add `access` tags to all routes by default?
+Each authorization check that we perform involves a round-trip to Elasticsearch, so they are not as cheap as we'd like. Therefore, we want to keep the number of authorization checks we perform within a single route to a minimum.
+Adding an `access` tag to routes that leverage the Saved Objects Service would be redundant in most cases, since the Saved Objects Service will be performing authorization anyway.
+
+
+## The `kibana_system` user
+
+The Kibana server authenticates to Elasticsearch using the `elastic/kibana` [service account](https://www.elastic.co/guide/en/elasticsearch/reference/current/service-accounts.html#service-accounts-explanation). This service account has privileges that are equivilent to the `kibana_system` reserved role, whose descriptor is managed in the Elasticsearch repository ([source link](https://github.com/elastic/elasticsearch/blob/56633e0a2934b03090ab726c33e5289129c98bfe/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java#L682)).
```

Comment:
__nit:__ we might need to update the URL once https://github.com/elastic/elasticsearch/pull/97603 lands (it might still tied to a specific commit, but it'd be easier to switch to `main` to see the latest version).

Reply:
Thanks for calling this out, I'll wait to merge this until https://github.com/elastic/elasticsearch/pull/97603 lands

Reply:
Done in https://github.com/elastic/kibana/pull/161697/commits/2671817c1dc7aa84a42081100bdb576a8fd17cae